src/Hosting.CommandLine/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using McMaster.Extensions.CommandLineUtils;
+using McMaster.Extensions.Hosting.CommandLine.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Threading.Tasks;
+
+namespace McMaster.Extensions.Hosting.CommandLine
+{
+    /// <summary>
+    /// Extension methods for <see cref="IHostBuilder"/> support.
+    /// </summary>
+    /// <seealso href="https://github.com/natemcmaster/CommandLineUtils/issues/134">host support</seealso>
+    public static class HostBuilderExtensions
+    {
+        /// <summary>
+        /// Runs <code>T</code> as a <see cref="CommandLineApplication"/> with the
+        /// supplied <code>args</code>.  This method should be the primary approach 
+        /// taken for command line applications.
+        /// </summary>
+        /// <typeparam name="T">The command line application implementation</typeparam>
+        /// <param name="hostBuilder">This instance</param>
+        /// <param name="args">The command line arguments</param>
+        /// <returns>A task whose result is the exit code of the application</returns>
+        public static async Task<int> RunCliAsync<T>(this IHostBuilder hostBuilder, string[] args) where T : class
+        {
+            using (var host = hostBuilder.UseCliLifetime().UseCli<T>(args).Build()) 
+            {
+                await host.StartAsync();
+                await host.WaitForShutdownAsync();
+                return ((CliLifetime)host.Services.GetService<IHostLifetime>()).ExitCode;
+            }
+        }
+
+        /// <summary>
+        /// Sets the <see cref="IHostLifetime"/> to one that handles running a 
+        /// <see cref="CommandLineApplication"/>.  It will ensure the
+        /// <see cref="ICliService"/> will be run to completion, the exit code gathered,
+        /// and the hosted application stopped.
+        /// <para>
+        /// This method is not intended to be used directly.  Instead, you should use
+        /// <see cref="RunCliAsync{T}(IHostBuilder, string[])"/>.
+        /// </para>
+        /// </summary>
+        /// <param name="hostBuilder">This instance</param>
+        /// <returns>This instance</returns>
+        public static IHostBuilder UseCliLifetime(this IHostBuilder hostBuilder)
```

Comment:
Can you provide a sample of how someone might use this? I'm having a hard time of thinking of a good use case. If we don't have one, I'd rather make this internal for now. It's always to give more API once it's had time to bake.

Reply:
Good point!  I really only added it as an analog to `UseConsoleLifetime`, but given that `UseCliLifetime` required `ICliService`, it makes sense to just [combine `UseCli` and `UseCliLifetime`](https://github.com/natemcmaster/CommandLineUtils/pull/167/commits/51b186261d0d79d2e51ca6742d4aa570b1d6b072).